lib/IRGen/IRGen.cpp

```diff
@@ -231,6 +231,37 @@ static void populatePGOOptions(std::optional<PGOOptions> &Out,
     return;
   }
 
+  if (Opts.EnableCSIRProfileGen) {
+    const bool hasUse = !Opts.UseProfile.empty();
+    Out = PGOOptions(
+      /*ProfileFile=*/ hasUse ? Opts.UseProfile : "",
+      /*CSProfileGenFile=*/ Opts.CSProfileGenFile,
+      /*ProfileRemappingFile=*/ "",
+      /*MemoryProfile=*/ "",
+      /*FS=*/ llvm::vfs::getRealFileSystem(),
+      /*Action=*/ hasUse ? PGOOptions::IRUse : PGOOptions::NoAction,
+      /*CSPGOAction=*/ PGOOptions::CSIRInstr,
+      /*ColdType=*/ PGOOptions::ColdFuncOpt::Default,
+      /*DebugInfoForProfiling=*/ Opts.DebugInfoForProfiling
+    );
+    return;
+  }
+
+  if (Opts.EnableIRProfileGen) {
```

Comment:
Can you add cases for `EnableIRProfileUse` and `EnableCSIRProfileUse`?

Reply:
Yes, my initial thought was to handle the IR-only usage modes in a follow-up PR, but I’m happy to include them here if that’s preferable. I was leaning that way since Clang uses `IndexedInstrProfReader` to decide between IR and CSIR profiles and doesn't have a dedicated input flag which takes in intent ir/cs usage. 

Would you recommend taking a similar approach in Swift as part of this PR?

[clang/lib/Frontend/CompilerInvocation.cpp#L1497](https://github.com/llvm/llvm-project/blob/44b7abcc75b005ab87e11e2beac155bf0b155992/clang/lib/Frontend/CompilerInvocation.cpp#L1497)

Reply:
Actually sorry I forgot there is no -fcs-profile-use. We can leave -profile-use to a separate PR if you want.

Reply:
No worries at all and thanks for pointing it out, makes sense to include `-ir-profile-use` in this PR. 👍